infrastructure/tournament-util/challonge_pubber.py

```diff
@@ -0,0 +1,82 @@
+# Requires achallonge, and _not_ pychal. Make sure to `pip uninstall pychal`, `pip install achallonge` etc before using.
+
+# Usage:
+# python challonge_pubber.py argv
+# argv[1]: a json produced by running the tournmanet
+# argv[2]: the challonge match number (as shown on the bracket page) of the first Challonge match (or, if argv[3] not specified, the only challonge match) whose result is to be published
+# argv[3]: optional; the challonge match number of the last Challonge match to be published, _exclusive_.
+
+# IMPORTANT -- Before running this:
+# Ensure you have achallonge, and not pychal, installed as Python packages.
+# Get the Challonge API Key, substitute it for API_KEY below. DON'T PUSH IT!
+# Get the tournament url, it's the alphanumeric string at the end of the tournament website's url. (e.g. http://challonge.com/thispart)
+# Get the lowest Challonge match id (see some commented code for an example). Set it to lowest_id.
+# Ensure the tournament is started and attachments are allowed (see some commented code for more info).
+
+import sys, json, challonge, asyncio
+
+replay_file_name = sys.argv[1]
+match_no_start = int(sys.argv[2])
+try:
+    match_no_end = int(sys.argv[3])
+except:
+    match_no_end = match_no_start+1
+
+
+async def run():
+    with open(replay_file_name, 'r') as replay_file:
+        replays = json.load(replay_file)
+        user = await challonge.get_user('mitbattlecode','API_KEY')
```

Comment:
can we make `API_KEY`, `url`, `lowest_id`, etc either environment variables, command-line arguments or stdin inputs? would help to avoid unintentional pushes

Reply:
good idea -- thanks!